iocore/net/SSLNetVConnection.cc

```diff
@@ -736,7 +736,7 @@ SSLNetVConnection::net_read_io(NetHandler *nh, EThread *lthread)
     break;
   case SSL_READ_ERROR:
     this->read.triggered = 0;
-    readSignalError(nh, static_cast<int>(r));
+    readSignalError(nh, (errno) ? errno : -ENET_SSL_FAILED);
```

Comment:
What abut this one? I guess it's checking errno set by `ssl_read_from_net`, but it can be changed in `readSignalAndUpdate`. I'm not opposed to propagating errno but we should secure the value right after calling a function.

Reply:
Afraid I don't understand the comment.  We are passing the errno that would have been generated from ssl_read_from_net.  It may change during the call to readSignalError, but the argument will be the value from the call point.  Would you like me to save the errno value directly after the all to ssl_read_from_net?

Reply:
> Would you like me to save the errno value directly after the all to ssl_read_from_net?

Not sure if it has to be all, but yes. A problem is that `errno` could be changed if I added some code between `ssl_read_from_net` (or `readSignalError`) and this line. If that's another I/O related code then it may be ok, but it might be something for completely different things (e.g. debugging, logging, etc).

So the current code is fine. It just looks fragile.

Reply:
pushed a commit to introduce a local var to keep the ssl read errno.